lib/source/defender/aws_iot_defender_api.c

```diff
@@ -179,26 +196,24 @@ AwsIotDefenderError_t AwsIotDefender_Start( AwsIotDefenderStartInfo_t * pStartIn
 
         if( metricsMutexCreateSuccess )
         {
-            /* Create disconnect job. */
-            taskPoolError = IotTaskPool_CreateJob( _disconnectRoutine, NULL, &_disconnectJobStorage, &_disconnectJob );
-            AwsIotDefender_Assert( taskPoolError == IOT_TASKPOOL_SUCCESS );
+            /*  Subscribe to Metrics Publish topic */
+            _metricsSubscribeRoutine();
 
-            /* Create metrics job. */
+            /* Create metrics publish job, which will periodically publish metrics */
             taskPoolError = IotTaskPool_CreateJob( _metricsPublishRoutine, NULL, &_metricsPublishJobStorage, &_metricsPublishJob );
+            /* Silence warnigns when asserts are disabled. */
+            ( void ) taskPoolError;
```

Comment:
If the `taskPoolError` value is not consumed at all, then it can just be removed. Instead every `IotTaskPool_*()` API call be `(void)`_ed_ to improve expressivity